sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/SimplifyConditionalSuite.scala

```diff
@@ -35,18 +35,24 @@ class SimplifyConditionalSuite extends PlanTest with PredicateHelper {
   }
 
   protected def assertEquivalent(e1: Expression, e2: Expression): Unit = {
-    val correctAnswer = Project(Alias(e2, "out")() :: Nil, OneRowRelation()).analyze
-    val actual = Optimize.execute(Project(Alias(e1, "out")() :: Nil, OneRowRelation()).analyze)
+    val correctAnswer = Project(Alias(e2, "out")() :: Nil, LocalRelation('a.int)).analyze
+    val actual = Optimize.execute(
+      Project(Alias(e1, "out")() :: Nil, LocalRelation('a.int)).analyze)
     comparePlans(actual, correctAnswer)
   }
 
   private val trueBranch = (TrueLiteral, Literal(5))
   private val normalBranch = (NonFoldableLiteral(true), Literal(10))
   private val unreachableBranch = (FalseLiteral, Literal(20))
-  private val nullBranch = (Literal.create(null, NullType), Literal(30))
-
-  val isNotNullCond = IsNotNull(UnresolvedAttribute(Seq("a")))
-  val isNullCond = IsNull(UnresolvedAttribute("b"))
+  private val nullValue = Literal.create(null, IntegerType)
```

Comment:
Why did you change from `NullType` to `IntegerType` here?

Reply:
I need the same dataType as i have for the `a` attribute. But i can just add another nullValue to the test and keep the previous with the original dataType.

Reply:
Yea, I think its better to avoid the behaviour changes in the existing tests.